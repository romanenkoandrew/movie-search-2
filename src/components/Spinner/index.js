import React from 'react';
import { Spin, Modal } from 'antd';
import PropTypes from 'prop-types';

const Spinner = ({ isLoading }) => {
  return (
    <Modal
      footer={null}
      visible={isLoading}
      centered
      closable={false}
      maskStyle={{ backgroundColor: 'rgba(0, 0, 0, 0.15)' }}
      width={0}
    >
      <Spin size='large' />
    </Modal>
  );
};

Spinner.propTypes = {
  isLoading: PropTypes.bool.isRequired,
};

export default Spinner;
