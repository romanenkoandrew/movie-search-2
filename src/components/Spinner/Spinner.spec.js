import React from 'react';
import { shallow } from 'enzyme';
import Spinner from './index';

describe('Spinner component', () => {
  it('should render Spinner component', () => {
    const component = shallow(<Spinner isLoading />);
    expect(component).toMatchSnapshot();
  });
});
