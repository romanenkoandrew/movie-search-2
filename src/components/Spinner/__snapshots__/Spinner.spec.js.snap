// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner component should render Spinner component 1`] = `
<Modal
  centered={true}
  closable={false}
  confirmLoading={false}
  footer={null}
  maskStyle={
    Object {
      "backgroundColor": "rgba(0, 0, 0, 0.15)",
    }
  }
  maskTransitionName="fade"
  okType="primary"
  transitionName="zoom"
  visible={true}
  width={0}
>
  <Spin
    size="large"
    spinning={true}
    wrapperClassName=""
  />
</Modal>
`;
